src/EmailAccountRecovery.sol

```diff
@@ -0,0 +1,217 @@
+                (recoveredAccount, accountSalt, verifierInitData)
+            )
+        );
+        return address(proxy);
+    }
+
+    /// @notice Handles an acceptance by a new guardian.
+    /// @dev This function validates the email auth message, deploys a new EmailAuth contract as a
+    /// proxy if validations pass and initializes the contract.
+    /// @param guardianVerifierImplementation The address of the guardian verifier implementation.
+    /// @param recoveredAccount The address of the account to be recovered.
+    /// @param accountSalt A bytes32 salt value used to ensure the uniqueness of the deployed guardian verifier.
+    /// @param verifierInitData The initialization data for the guardian verifier.
+    /// @param proofData The proof data required for the guardian verifier.
+    /// the command in the given email auth message.
+    function handleAcceptance(
+        address guardianVerifierImplementation,
+        address recoveredAccount,
+        bytes32 accountSalt,
+        bytes memory verifierInitData,
+        IGuardianVerifier.ProofData memory proofData
+    ) external {
+        address guardian = computeGuardianVerifierAddress(
+            guardianVerifierImplementation,
+            recoveredAccount,
+            accountSalt,
+            verifierInitData
+        );
+
+        if (guardian.code.length == 0) {
+            address proxyAddress = deployGuardianVerifierProxy(
+                guardianVerifierImplementation,
+                recoveredAccount,
+                accountSalt,
+                verifierInitData
+            );
+
+            if (proxyAddress != guardian) {
+                revert InvalidProxyDeployment();
+            }
+        }
+
+        bool isVerified = IGuardianVerifier(guardian).verifyProofStrict(
+            recoveredAccount,
+            proofData
+        );
+
+        acceptGuardian(guardian, recoveredAccount);
+    }
+
+    /// @notice Processes the recovery based on an email from the guardian.
+    /// @dev Verify the provided email auth message for a deployed guardian's EmailAuth contract and
+    /// a specific command template for recovery.
+    /// Requires that the guardian is already deployed
+    /// @param guardian The address of the guardian who is processing the recovery request
+    /// @param recoveredAccount The address of the account to be recovered.
+    /// @param accountSalt A bytes32 salt value used to ensure the uniqueness while deploying the guardian verifier
+    /// @param recoveryDataHash The hash of the recovery data
+    /// @param proofData The proof data for the guardian verifier.
+    function handleRecovery(
+        address guardian,
+        address recoveredAccount,
+        bytes32 accountSalt,
+        bytes32 recoveryDataHash,
+        IGuardianVerifier.ProofData memory proofData
+    ) external {
+        // TODO: Do we need to compute the guardian in handleRecovery as well, or just take the address as an input
```

Comment:
The reason we compute things in these functions is to align it with the payload the relayer expects for calls related to the `EmailAuth` circuits. For these new contracts, it's safe to break from this interface and way of doing things.

So we can just pass the required arguments if they are sufficiently generic for all of the verifiers we want to support